ui/apps/platform/src/Containers/Policies/PatternFly/Modal/EnableDisableNotificationModal.tsx

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+import { TableComposable, Thead, Tbody, Tr, Th, Td } from '@patternfly/react-table';
+import pluralize from 'pluralize';
+
+import { enableDisableNotificationsForPolicies } from 'services/PoliciesService';
+import ConfirmationModal from 'Components/PatternFly/ConfirmationModal';
+import { AlertVariantType } from 'hooks/patternfly/useToasts';
+import useTableSelection from 'hooks/useTableSelection';
+import { NotifierIntegration } from 'types/notifier.proto';
+
+export type EnableDisableType = 'enable' | 'disable' | '';
+
+type EnableDisableNotificationModalProps = {
+    enableDisableType: EnableDisableType;
+    setEnableDisableType: (type) => void;
+    fetchPoliciesHandler: () => void;
+    addToast: (text: string, variant: AlertVariantType, content?: string) => void;
+    selectedPolicyIds: string[];
+    notifiers: NotifierIntegration[];
+};
+
+function EnableDisableNotificationModal({
+    enableDisableType,
+    setEnableDisableType,
+    fetchPoliciesHandler,
+    addToast,
+    selectedPolicyIds,
+    notifiers,
+}: EnableDisableNotificationModalProps) {
+    // Handle selected rows in table
+    const { selected, allRowsSelected, onSelect, onSelectAll, onClearAll, getSelectedIds } =
+        useTableSelection(notifiers);
+
+    const selectedNotifierIds = getSelectedIds();
+
+    function enableDisableNotificationHandler() {
+        const selectedNotifiers =
+            notifiers.length === 1 ? notifiers.map((notifier) => notifier.id) : selectedNotifierIds;
+        return enableDisableNotificationsForPolicies(
+            selectedPolicyIds,
+            selectedNotifiers,
+            enableDisableType === 'disable'
+        )
+            .then(() => {
+                fetchPoliciesHandler();
+                addToast(`Successfully ${enableDisableType}d notification`, 'success');
+            })
+            .catch(({ response }) => {
+                addToast(
+                    `Could not ${enableDisableType} notification`,
+                    'danger',
+                    response.data.message
+                );
+            });
+    }
+
+    function onConfirmEnableDisableNotifications() {
+        setEnableDisableType('');
+        enableDisableNotificationHandler().finally(() => {});
+    }
+
+    function onCancelEnableDisableNotifications() {
+        setEnableDisableType('');
+        onClearAll();
+    }
+
+    return (
+        <ConfirmationModal
+            ariaLabel={`Confirm ${enableDisableType} notification`}
+            confirmText={enableDisableType}
+            title={`${enableDisableType} notifications`}
```

Comment:
Can we use `capitalize(enableDisableType)` from lodash instead of CSS style rules?
* So modal title will have sentence case **Enable notifications** like menu items
* Also reduces potential for confusion when writing unit tests
* Because style rules affect all existing occurrences of `ConfirmationModal` element

Reply:
Is there a reason why the menu items are singular **notification** and the modal title is plural **notifications**

Reply:
oh no that was unintentional. let me make them all singular.